modules/store/src/selector.ts

```diff
@@ -120,81 +120,122 @@ export function defaultMemoize(
 export function createSelector<State, S1, Result>(
   s1: Selector<State, S1>,
   projector: (s1: S1) => Result
-): MemoizedSelector<State, Result>;
+): MemoizedSelector<State, Result, (s1: S1) => Result>;
 export function createSelector<State, Props, S1, Result>(
   s1: SelectorWithProps<State, Props, S1>,
   projector: (s1: S1, props: Props) => Result
-): MemoizedSelectorWithProps<State, Props, Result>;
+): MemoizedSelectorWithProps<
+  State,
+  Props,
+  Result,
+  (s1: S1, props: Props) => Result
+>;
 export function createSelector<State, S1, Result>(
   selectors: [Selector<State, S1>],
   projector: (s1: S1) => Result
-): MemoizedSelector<State, Result>;
+): MemoizedSelector<State, Result, (s1: S1) => Result>;
 export function createSelector<State, Props, S1, Result>(
   selectors: [SelectorWithProps<State, Props, S1>],
   projector: (s1: S1, props: Props) => Result
-): MemoizedSelectorWithProps<State, Props, Result>;
+): MemoizedSelectorWithProps<
+  State,
+  Props,
+  Result,
+  (s1: S1, props: Props) => Result
+>;
 
-export function createSelector<State, S1, S2, Result>(
+export function createSelector<
+  State,
+  S1,
+  S2,
+  Result,
+  Projector extends (s1: S1, s2: S2) => Result
```

Comment:
?

Reply:
Sorry, it was a leftover from my playing around to come up with something more wise but you nailed it with `typeof projector`. 👏